apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/StorageFixture.cs

```diff
@@ -242,9 +249,16 @@ void CreateObject()
 
         }
 
-        internal Bucket CreateBucket(string name, bool multiVersion)
+        internal Bucket CreateBucket(string name, bool multiVersion, bool softDelete = false)
         {
-            var bucket = Client.CreateBucket(ProjectId, new Bucket { Name = name, Versioning = new Bucket.VersioningData { Enabled = multiVersion } });
+            var bucket = Client.CreateBucket(ProjectId,
+                new Bucket
+                {
+                    Name = name,
+                    Versioning = new Bucket.VersioningData { Enabled = multiVersion },
+                    // The minimum allowed for soft delete is 7 days.
+                    SoftDeletePolicy = softDelete ? new Bucket.SoftDeletePolicyData { RetentionDurationSeconds = 7*24*60*60 } : null,
```

Comment:
Nit: spaces between the `7*24*60*60` - or better, work that out earlier:

```csharp
var sevenDaysInSeconds = (int) TimeSpan.FromDays(7).TotalSeconds;
```

Reply:
(Or preferably NodaConstants.SecondsPerWeek, but hey... one day I'll justify a NodaTime dependency ;)